src/main/java/at/helpch/placeholderapi/expansion/player/PlayerExpansion.java

```diff
@@ -0,0 +1,524 @@
+
+    @Override
+    public @NotNull String getIdentifier() {
+        return "player";
+    }
+
+    @Override
+    public @NotNull String getAuthor() {
+        return "HelpChat";
+    }
+
+    @Override
+    public @NotNull String getVersion() {
+        return "3.0.0";
+    }
+
+    @Override
+    public Map<String, Object> getDefaults() {
+        return ImmutableMap.<String, Object>builder()
+            .put("direction.north", "N")
+            .put("direction.north_east", "NE")
+            .put("direction.east", "E")
+            .put("direction.south_east", "SE")
+            .put("direction.south", "S")
+            .put("direction.south_west", "SW")
+            .put("direction.west", "W")
+            .put("direction.north_west", "NW")
+            .put("ping.formatting.>200", "&4")
+            .put("ping.formatting.>150", "&c")
+            .put("ping.formatting.>100", "&e")
+            .put("ping.formatting.>75", "&2")
+            .put("ping.formatting.>50", "&a")
+            .put("ping.formatting.>25", "&b")
+            .put("ping.formatting.=0", "&9")
+            .build();
+    }
+
+    @Override
+    public boolean canRegister() {
+        pingFormatter = new PingFormatter(getConfigSection("ping.formatting"));
+        directionNames = Optional.ofNullable(getConfigSection("direction"))
+            .map(section -> section.getValues(false))
+            .orElseGet(HashMap::new);
+        return true;
+    }
+
+    /**
+     * Format a {@code boolean} as {@link PlaceholderAPIPlugin#booleanTrue()} or {@link PlaceholderAPIPlugin#booleanFalse()}
+     *
+     * @param bool boolean
+     * @return {@link PlaceholderAPIPlugin#booleanTrue()} if the boolean is true, otherwise {@link PlaceholderAPIPlugin#booleanFalse()}
+     */
+    private @NotNull String bool(final boolean bool) {
+        return bool ? PlaceholderAPIPlugin.booleanTrue() : PlaceholderAPIPlugin.booleanFalse();
+    }
+
+    private @NotNull String getItemInfo(@Nullable final ItemStack item, @NotNull final String args) {
+        switch (args) {
+            case "_name":
+                return ItemUtil.getName(item);
+            case "_data":
+                return String.valueOf(ItemUtil.getData(item));
+            case "_durability":
+                return String.valueOf(ItemUtil.getDurability(item));
+            default:
+                return ItemUtil.getMaterialName(item);
+        }
+    }
+
+    @SuppressWarnings("SpellCheckingInspection")
+    @Override
+    public @Nullable String onRequest(OfflinePlayer offlinePlayer, @NotNull String params) {
+        final boolean targetedPing = params.startsWith("ping_");
+        final boolean targetedColoredPing = params.startsWith("colored_ping_");
+
+        if (targetedPing || targetedColoredPing) {
+            final String targetName = params.substring((targetedPing ? "ping_" : "colored_ping_").length());
+            return Optional.ofNullable(Bukkit.getPlayer(targetName))
```

Comment:
Should we use getPlayerExact? Doesn't seem very important for this specific placeholder though.